helm/router/templates/ingress.yaml

```diff
@@ -0,0 +1,54 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := .Release.Name -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
```

Comment:
These were generated by `helm create`, I don't know if we want to keep these different scenarios or just assume that we will be using k8s >= 1.19

Reply:
We don't need to care about k8s <= 1.19, because using any version more than three releases old is a bad idea anyway because of the [version skew policy](https://kubernetes.io/releases/version-skew-policy/#supported-version-skew). I'd probably lean towards cleaning up these conditionals but I don't mind.

Reply:
leaving it for now